frontend/src/pages/GraphPF/styles/styleEdge.tsx

```diff
@@ -1,23 +1,128 @@
-import { Edge, observer, ScaleDetailsLevel, WithSelectionProps } from '@patternfly/react-topology';
-import { BaseEdge } from '../components/edge';
+import { DefaultEdge, Edge, observer, ScaleDetailsLevel, WithSelectionProps } from '@patternfly/react-topology';
 import useDetailsLevel from '@patternfly/react-topology/dist/esm/hooks/useDetailsLevel';
+import { PFColors } from 'components/Pf/PfColors';
 import * as React from 'react';
 import { kialiStyle } from 'styles/StyleUtils';
+import { classes } from 'typestyle';
 
-// This is the registered Edge component override that utilizes our customized Edge.tsx component.
+// This is our styled edge component registered in stylesComponentFactory.tsx.  It is responsible for adding our custom customizations that get passed down to DefaultEdge.  The current customizations:
+//   data.pathStyle?: React.CSSProperties // additional CSS stylings for the edge/path (not the endpoint).
+//   data.isFind?: boolean                // adds graph-find overlay
+//   data.isUnhighlighted?: boolean       // adds unhighlight effects
+//   data.hasSpans?: Span[]               // adds trace overlay
+//   add showTag prop and show scaled tag on hover (when showTag is false)
+
+const ColorFind = PFColors.Gold400;
+const ColorSpan = PFColors.Purple200;
+const OverlayOpacity = 0.3;
+const OverlayWidth = 30;
 
 type StyleEdgeProps = {
   element: Edge;
 } & WithSelectionProps;
 
-const tagClass = kialiStyle({
-  fontFamily: 'Verdana,Arial,Helvetica,sans-serif,pficon'
-});
-
 const StyleEdgeComponent: React.FC<StyleEdgeProps> = ({ element, ...rest }) => {
   const data = element.getData();
   const detailsLevel = useDetailsLevel();
 
+  let cssClasses: string[] = [];
+
+  const onMouseEnter = () => {
+    data.onHover(element, true);
+  };
+
+  const onMouseLeave = () => {
+    data.onHover(element, false);
+  };
+
+  // Change edge color according to the pathStyle
+  const edgeClass = kialiStyle({
+    $nest: {
+      '.pf-topology__edge__link': data.pathStyle
```

Comment:
According to `typestyle` documentation (https://typestyle.github.io/#/core), it is best practice to add symbol `$` to overwrite third party nested CSS classes:

```
const container = style({
  $nest: {
    '& .third-party-class': {
      color: 'red'
    }
  }
});
```
Most of the times it works fine without the `&` symbol, but not in all cases (if the third party CSS class has more specifity than yours). More information here: https://github.com/kiali/kiali/pull/6562